tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,325 @@
+            ),
+            # if the number of lines of code is not within the specified range,
+            # discard the document
+            ScoreFilter(
+                NumberOfLinesOfCodeFilter(min_lines=5, max_lines=20000),
+                text_field="text",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+def redact_pii(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redacts personally identifiable information (PII) from extracted comment lines in a given dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents with PII.
+
+    Returns:
+        DocumentDataset: The redacted dataset with PII replaced by a generic value.
+    """
+    DEFAULT_MAX_DOC_SIZE = 3000000
+    redactor = Modify(
+        PiiModifier(
+            supported_entities=[
+                "PERSON",
+                "EMAIL_ADDRESS",
+            ],
+            anonymize_action="replace",
+            device="gpu",
+        ),
+        text_field="extracted_comment",
+    )
+    return redactor(dataset)
+
+
+def redact_code(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redact PII on comment lines on a  given code dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset to be redacted.
+
+    Returns:
+        DocumentDataset: The redacted dataset.
+    """
+
+    # functions to extract comment lines from each row in a dataframe
+    def func(row):
+        return row["text"][row["text"].find("/*") : row["text"].find("*/") + 2]
+
+    def func2(row):
+        comment = row["text"][row["text"].find("/*") : row["text"].find("*/") + 2]
+        return row["text"].replace(comment, str(row["extracted_comment"]))
+
+    dataset.df["extracted_comment"] = dataset.df.apply(func, axis=1, meta=(None, str))
+    redacted_dataset = redact_pii(dataset)
+    redacted_dataset.df["text"] = redacted_dataset.df.apply(
+        func2, axis=1, meta=(None, str)
+    )
+    redacted_dataset.df = redacted_dataset.df.drop(["extracted_comment"], axis=1)
+
+    return redacted_dataset
+
+
+def dedupe(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Remove exact duplicates from the given DocumentDataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents.
+
+    Returns:
+        DocumentDataset: The deduplicated dataset.
+    """
+    deduplicator = ExactDuplicates(id_field="id", text_field="text", hash_method="md5")
+    # Find the duplicates
+    duplicates = deduplicator(dataset)
+    docs_to_remove = duplicates.df.map_partitions(
+        lambda x: x[x._hashes.duplicated(keep="first")]
+    )
+    # Remove the duplicates using their IDs.
+    duplicate_ids = list(docs_to_remove.compute().id)
+    dataset_df = dataset.df
+    deduped = dataset_df[~dataset_df.id.isin(duplicate_ids)]
+    return DocumentDataset(deduped)
+
+
+class LineCountFilter_text(DocumentFilter):
```

Comment:
Nit: Make class casing consistent. `LineCountFilter_text ` -> `TextLineCountFilter`.

Reply:
Its already consistent right? I dont see this in other tutorials

Reply:
Class names should be consistent in style with other class names in NeMo Curator and the [Python style](https://peps.python.org/pep-0008/#class-names) for class names. That means no underscores in the class name, and it should follow the CapWords convention.

Reply:
Got it. Makes sense :) Corrected it